pkg/nodeconfig/payload/payload.go

```diff
@@ -80,7 +80,36 @@ const (
 	// TODO: This script is doing both CNI configuration and HNS endpoint creation, two things that aren't necessarily
 	//       related. Correct that in: https://issues.redhat.com/browse/WINC-882
 	// networkConfTemplate is the template used to generate the network configuration script
-	networkConfTemplate = `# This script ensures the contents of the CNI config file is correct, and returns the HNS endpoint IP
+	networkConfTemplate = `# This script ensures the contents of the CNI config file is correct, and creates the kube-proxy config file.
+
+param(
+    [string]$hostnameOverride,
+    [string]$clusterCIDR,
+    [string]$kubeConfigPath,
+    [string]$kubeProxyConfigPath,
+    [string]$verbosity
+)
+  # this compares the config with the existing config, and replaces if necessary
+  function Compare-And-Replace-Config {
+    param (
+        [string]$ConfigPath,
+        [string]$NewConfigContent
+    )
+    
+    # Read existing config content
+    $existing_config = ""
+    if (Test-Path -Path $ConfigPath) {
+        $config_file_content = Get-Content -Path $ConfigPath -Raw
+        if ($config_file_content -ne $null) {
+` + "        $existing_config=$config_file_content.Replace(\"`r\",\"\")" + `
```

Comment:
>  $existing_config=$config_file_content.Replace(\"`r\",\"\")"

Is this replacement taking place in PowerShell? PowerShell uses 
```
`n`r
```
 for new lines on Windows.

Consider using the `[Environment]::NewLine`  resolver to easy to char escaping

Reply:
I follow, @wgahnagl. What is the goal here? To replace all the `new line` with the empty string `""`?

Does the `Replace` function replaces all occurrences?

Reply:
Please don't change any of the replace logic with this ticket. False positives on this were the cause of a major networking bug. There is no easy way to test this logic given it is in powershell.